src/sentry/api/endpoints/organization_transaction_anomaly_detection.py

```diff
@@ -0,0 +1,92 @@
+from datetime import datetime
+
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.bases import OrganizationEventsEndpointBase
+from sentry.api.utils import get_date_range_from_params
+from sentry.search.events.builder import QueryBuilder
+from sentry.snuba.dataset import Dataset
+
+
+class OrganizationTransactionAnomalyDetectionEndpoint(OrganizationEventsEndpointBase):
+    def get(self, request: Request, organization) -> Response:
+        start, end = get_date_range_from_params(request.GET)
+        query_start, query_end, granularity = self.map_snuba_queries(
+            start.timestamp(), end.timestamp()
+        )
+
+        # remove unused and overwrite relevant time params
+        for param in ["statsPeriod", "statsPeriodStart", "statsPeriodEnd"]:
+            request.GET.pop(param, None)
+        request.GET[start] = query_start
+        request.GET[end] = query_end
+
+        params = self.get_snuba_params(request, organization)
+        query = request.GET.get("query")
+        snql_query = self.get_snuba_query(params, query, granularity)
+
+        return self.get_anomalies(snql_query)
+
+    @staticmethod
+    def map_snuba_queries(start, end):
```

Comment:
There's already `sentry.utils.dates.get_rollup_from_request` which does a very similar functionality to this, not sure if anomalies needs custom intervals that differ from the ones we already use

Reply:
yeah, at least for now we want to be able to adjust this separately (maybe we can use that instead eventually)